server/src/main/kotlin/suwayomi/tachidesk/global/impl/util/Jwt.kt

```diff
@@ -0,0 +1,97 @@
+package suwayomi.tachidesk.global.impl.util
+
+import com.auth0.jwt.JWT
+import com.auth0.jwt.JWTVerifier
+import com.auth0.jwt.algorithms.Algorithm
+import com.auth0.jwt.exceptions.JWTVerificationException
+import io.github.oshai.kotlinlogging.KotlinLogging
+import suwayomi.tachidesk.server.user.UserType
+import java.security.SecureRandom
+import java.time.Instant
+import javax.crypto.spec.SecretKeySpec
+import kotlin.io.encoding.Base64
+import kotlin.io.encoding.ExperimentalEncodingApi
+import kotlin.time.Duration.Companion.days
+import kotlin.time.Duration.Companion.hours
+
+object Jwt {
+    private val logger = KotlinLogging.logger {}
+
+    private const val ALGORITHM = "HmacSHA256"
+    private val accessTokenExpiry = 1.hours
+    private val refreshTokenExpiry = 60.days
+    private const val ISSUER = "tachidesk"
+    private const val AUDIENCE = "" // todo audience
+
+    @OptIn(ExperimentalEncodingApi::class)
+    fun generateSecret(): String {
+        val keyBytes = ByteArray(32)
+        SecureRandom().nextBytes(keyBytes)
+
+        val secretKey = SecretKeySpec(keyBytes, ALGORITHM)
+
+        return Base64.encode(secretKey.encoded)
+    }
+
+    private val algorithm: Algorithm = Algorithm.HMAC256(generateSecret()) // todo store secret
+    private val verifier: JWTVerifier = JWT.require(algorithm).build()
+
+    class JwtTokens(
+        val accessToken: String,
+        val refreshToken: String,
+    )
+
+    fun generateJwt(): JwtTokens {
+        val accessToken = createAccessToken()
+        val refreshToken = createRefreshToken()
+
+        return JwtTokens(
+            accessToken = accessToken,
+            refreshToken = refreshToken,
+        )
+    }
+
+    fun refreshJwt(refreshToken: String): JwtTokens {
+        val jwt = verifier.verify(refreshToken)
+        require(jwt.getClaim("token_type").asString() == "refresh") {
+            "Cannot use access token to refresh"
+        }
+        return generateJwt()
```

Comment:
```suggestion
    fun refreshJwt(refreshToken: String): String {
        val jwt = verifier.verify(refreshToken)
        require(jwt.getClaim("token_type").asString() == "refresh") {
            "Cannot use access token to refresh"
        }
        return createAccessToken()
```
Lets only refresh the access token, I want the refresh token to only be given once per login.
You will need to change the refreshToken mutation as well

Reply:
Makes sense, I think that's the typical usecase